weather.py

```diff
@@ -1,104 +1,426 @@
+    > return iso_string.strftime(f"%A %d %B %Y")
+
+#curious about: input = (input(write your year-month-day)) > if year-month-day is 2001-12-29 > the date value below will read the year-month-date and this will be the input into the function. Code for this below:
+    > iso_string = input("write your date in year/month/day eg. 2001/12/29 ")
+    > print(convert_date(iso_string))
+# '''
+
+
+### CHALLENGE 2
+def convert_f_to_c(temp_in_farenheit):
+    """Converts an temperature from farenheit to celcius.
+
+    Args:
+        temp_in_farenheit: float representing a temperature.
+    Returns:
+        A float representing a temperature in degrees celcius, rounded to 1dp.
+    """
+    x = (float(temp_in_farenheit) - 32) * 5/9
+    x = round(x,1)
+    return x
+
+#print(convert_f_to_c(90))
+
+'''
+Test cases to put into 'convert_f_to_c' to see if the function works 
+    1. 90 | expected result = 32.2
+    2. -10 | expected result = -23.3
+    3. 64.4 | expected_result = 18
+    4. "77" | expected_result = 25.0
+'''
+
+
+### CHALLENGE 3
+def calculate_mean(weather_data):
+    '''Calculates the mean value from a list of numbers.
+
+    Args:
+        weather_data: a list of numbers.
+    Returns:
+        A float representing the mean value.
+    '''
+
+    weather_data_str = []
+    for data in weather_data:
+        data = float(data)
+        weather_data_str.append(data)
+    # print(weather_data_str)
+
+    total = sum(weather_data_str)
+    quantity = len(weather_data_str)
+    return total/quantity
+
+#print(calculate_mean([-51, -58, -59, -52, -52, -48, -47, -53]))
+
+'''
+Test cases to put into 'calculate_mean' to see if the function works 
+    1. temperatures = [49, 57, 56, 55, 53] | expected_result = 54
+    2. temperatures = [51.0, 58.2, 59.9, 52.4, 52.1, 48.4, 47.8, 53.43] | expected_result = 52.90375
+    3. temperatures = ["51", "58", "59", "52", "52", "48", "47", "53"] | expected_result = 52.5
+    4. temperatures = [-51, -58, -59, -52, -52, -48, -47, -53] | expected_result = -52.5
+'''
+
+'''
+Annie's working notepad 
+    weather_data = [1,2,3,4]
+    total = sum(weather_data)
+    quantity = len(weather_data)
+    mean = total/quantity
+    print(total)
+    print(quantity)
+    print(mean)
+'''
+
+
+### CHALLENGE 4
+
+def load_data_from_csv(csv_file):
+
+    """Reads a csv file and stores the data in a list.
+
+    Args:
+        csv_file: a string representing the file path to a csv file.
+    Returns:
+        A list of lists, where each sublist is a (non-empty) line in the csv file.
+    """
+
+    csv_list = []
+    final_list = []
+    
+    with open(csv_file) as file:
+        reader = csv.reader(file)
+        next(reader)
+        for row in reader:
+            if not row:
+                continue
+            if row:
+                max = int(row[-1])
+                min = int(row[-2])
+                row[-1] = max
+                row[-2] = min
+                csv_list.append(row)

```

Comment:
Though this works, if we are modifying data, it is generally good practice to keep the modified data output separate from the input so that it's more readable.

E.g.
```suggestion
                new_row = [row[0], min, max]
                csv_list.append(new_row)
```